StreamChatClient_v3/APIClient/RequestEncoder.swift

```diff
@@ -0,0 +1,194 @@
+//
+// RequestEncoder.swift
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// On object responsible for creating a `URLRequest`, and encoding all required and `Endpoint` specific data to it.
+protocol RequestEncoder {
+    /// A delegate the encoder uses for obtaining the current `connectionId`.
+    ///
+    /// Trying to encode an `Endpoint` with the `requiresConnectionId` set to `true` without setting the delegate
+    var connectionIdProviderDelegate: ConnectionIdProviderDelegate? { get set }
+    
+    /// Asynchronously creates a new `URLRequest` with the data from the `Endpoint`. It also adds all required data
+    /// like an api key, etc.
+    ///
+    /// - Parameters:
+    ///   - endpoint: The `Endpoint` to be encoded.
+    ///   - completion: Called when the encoded `URLRequest` is ready. Called with en `Error` if the encoding fails.
+    func encodeRequest<ResponsePayload: Decodable>(for endpoint: Endpoint<ResponsePayload>,
+                                                   completion: @escaping (Result<URLRequest, Error>) -> Void)
+    
+    /// Creates a new `RequestEncoder`.
+    ///
+    /// - Parameters:
+    ///   - baseURL: The base URL for all requests.
+    ///   - apiKey: The app specific API key.
+    init(baseURL: URL, apiKey: APIKey)
+}
+
+/// The default implementation of `RequestEncoder`.
+struct DefaultRequestEncoder: RequestEncoder {
+    let baseURL: URL
+    let apiKey: APIKey
+    
+    weak var connectionIdProviderDelegate: ConnectionIdProviderDelegate?
+    
+    func encodeRequest<ResponsePayload: Decodable>(for endpoint: Endpoint<ResponsePayload>,
+                                                   completion: @escaping (Result<URLRequest, Error>) -> Void) {
+        var request: URLRequest
+        
+        do {
+            // Prepare the URL
+            var url = try encodeRequestURL(for: endpoint, queryItems: endpoint.queryItems)
```

Comment:
This looks a little bit weird. Why you can't get `queryItems` from `endpoint` inside `encodeRequestURL`? 
And you don't use it inside `encodeRequestURL`.
So, you removed regular `queryItems`. It looks like we have only `Encodable`.

Reply:
You absolutely right, this should just be `encodeRequestURL(for: endpoint)` because query items are set on L53. Thanks

Reply:
fixed in 4db432c5bf79beef846a21d8ce102e0927eca5da